src/backend/joanie/payment/backends/lyra/__init__.py

```diff
@@ -319,6 +319,35 @@ def create_zero_click_payment(self, order, installment, credit_card_token):
 
         return True
 
+    def is_already_debited(self, order, installment):
+        """
+        Check if the installment has already been paid.
+        If so, update the installment state to paid.
+        """
+        if installment["state"] == PAYMENT_STATE_PAID:
+            return True
+
+        url = f"{self.api_url}Order/Get"
+        payload = {
+            "orderId": str(order.id),
+        }
+        response_json = self._call_api(url, payload)
+        answer = response_json.get("answer")
+
+        if not answer:
+            return False
+
+        transactions = answer.get("transactions", [])
+        for transaction in transactions:
+            if transaction.get("metadata", {}).get("installment_id") == str(
+                installment["id"]
+            ):
+                installment["state"] = PAYMENT_STATE_PAID
+                order.save()
+                return True
```

Comment:
I think you should check the status of the transaction. I think you could get transaction in refused state.